### ISSUE TITLE
Upgrade to React v18.0

### ISSUE THREAD

**Issue by yonikid15** [GCTC-NTGC/gc-digital-talent#2404]
Upgrade guide: https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html

**Comment by petertgiles** [comment]
Make sure to update @types/react and @types/react-dom as well.

**Comment by vd1992** [comment]
Looking at this to poke around a bit... So far, issues I've noted

### **Children must be explicitly mentioned as a prop**
Easy enough fix, just add `children?: React.ReactNode;` into the type

### **`<IntlProvider>` type issues**
Does not accept it as a JSX element, dunno what the issue is

![image](https://user-images.githubusercontent.com/40485260/168403280-dd705a0a-cd42-471b-931d-2283471e1809.png)


### Won't accept styling functions on label.formatMessage
Type errors thrown, see the squiggly difference below. Deleting `Bold` fixes it

![image](https://user-images.githubusercontent.com/40485260/168403188-f9392609-cd33-410e-9766-493c65e4b6e6.png)

**Comment by vd1992** [comment]
update for reference 

![image](https://user-images.githubusercontent.com/40485260/168403471-ed30a06c-eb01-4975-b0b7-7f22c0b9185f.png)

**Comment by patcon** [comment]
Seems formatjs isn't compat with react 18 yet:
https://github.com/formatjs/formatjs/issues/3510#issuecomment-1094219578

Might be fixed through that link in definitelytyped by now.

**Comment by vd1992** [comment]
If it is a problem elsewhere then no point worrying about this now, I thought it was something weird on my end. 
Gonna throw this back for now then.

**Comment by petertgiles** [comment]
React-hook-form blocked at 7.33 on this.

https://github.com/GCTC-NTGC/gc-digital-talent/pull/3239

**Comment by petertgiles** [comment]
Reach blocked and 0.16 on this.
#3260

**Comment by mnigh** [comment]
Some of the previously mentioned blockers looked to be unblocked over time:

- formatjs supports react 18 [as well](https://github.com/formatjs/formatjs/commit/8808a95c3b377ee9b4e765ecfce74b45fd1c7700) (i think?)
- react-hook-form supports react 18 on the [version we are on](https://github.com/react-hook-form/react-hook-form/commit/fe6987b64a47ede54831d43196ac705b7736fee7#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L102).
- reach is no longer part of this project

🥳

**Comment by patcon** [comment]
patcon: testing-library founder says never use act. warning is a red herring.
peter: something more strict?

peter: pref for react 187 before next dependabot PRs
tristan: let's create new ticket for warnings. peter will try first

**Comment by patcon** [comment]
Ah, found the place in Slack where I summarized this and linked to Kent C Dodds saying stuff:
https://talent-cloud.slack.com/archives/C0259G6KXJ6/p1661803010061259

> I will go to my grave saying, 'you will NEVER need to wrap userEvent or fireEvent in act()', I will say that until I die -- Kent C Dodds
